README.md

```diff
@@ -28,14 +117,50 @@ db.put('name', 'encoding-down', function (err) {
 
 ### `const db = require('encoding-down')(db[, options])`
 
-* `db` `abstract-leveldown` compatible db such as `leveldown`, `memdown`, `level-js` etc
-* `options.keyEncoding` (string) defaults to `'utf8'`
-* `options.valueEncoding` (string) defaults to `'utf8'`
+* `db` must be an [`abstract-leveldown`] compliant store
+* `options` are passed to [`level-codec`]:
+  - `keyEncoding`: encoding to use for keys
+  - `valueEncoding`: encoding to use for values
+
+Both encodings default to `'utf8'`. They can be a string (builtin `level-codec` encoding) or an object (custom encoding).
+
+## Custom encodings
+
+Please refer to [`level-codec` documentation][encoding-format] for a precise description of the format. Here's a quick example with `level` and `async/await` just for fun:
 
-`options` are passed to [`level-codec`](https://github.com/level/codec), see [supported encodings](https://github.com/Level/codec#encodings) for more information.
+```js
+const level = require('level')
+const lexint = require('lexicographic-integer')
+
+async function main () {
+  const db = level('./db8', {
+    keyEncoding: {
+      // Hey, someone should publish this!
+      type: 'lexicographic-integer',
+      encode: (n) => lexint.pack(n, 'hex'),
+      decode: lexint.unpack,
```

Comment:
but haha, this doesn't work until #19 lands. Because if `unpack` receives a Buffer it won't parse it as hex.

Reply:
Should be ready for merge now with tests for `.keyAsBuffer` and `.valueAsBuffer`